Sources/Starknet/Providers/StarknetProviderProtocol.swift

```diff
@@ -122,11 +132,13 @@ public protocol StarknetProviderProtocol {
     ///  - transactions: list of transactions to simulate
     ///  - blockId: block used to run the simulation
     ///  - simulationFlags: a set of simulation flags
-    func simulateTransactions(_ transactions: [any StarknetTransaction], at blockId: StarknetBlockId, simulationFlags: Set<StarknetSimulationFlag>) async throws -> [StarknetSimulatedTransaction]
+    ///
+    ///  - Returns: array of simulated transactions
+    func simulateTransactions(_ transactions: [any StarknetExecutableTransaction], at blockId: StarknetBlockId, simulationFlags: Set<StarknetSimulationFlag>) async throws -> [StarknetSimulatedTransaction]
 }
 
 let defaultBlockId = StarknetBlockId.tag(.pending)
-
+let defaultSimulationFlagsForEstimateFee: Set<StarknetSimulationFlagForEstimateFee> = []
```

Comment:
In Account `estimateFee` functions, I think we should default to passing the flag, since I remember some nodes had problem validating signatures for txs with query versions. 

As for `Provider.estimateFee`, I'm not sure, as Provider methods should probably be more explicit. Might be better to send request with no flags by default, or remove default value for `simulation_flags` altogether, as in `simulateTransactions`.

WDYT?

Reply:
> In Account estimateFee functions, I think we should (for now) default to passing the .skipValidate flag, since, as I remember, some nodes had problem validating signatures for txs with query versions.

hmm, that's weird. was this checked with some newer nodes' releases? Do you happen to know if that was by design, or by mistake? Either way, I'm afraid we will set this up and not change it, ever. So (if that's by mistake) let's maybe pass on this idea, and push on nodes maintainers to fix this? If that's by design then you're right, and we can add this.

> As for Provider.estimateFee, I'm not sure, as Provider methods should probably be more explicit. Might be better to send request with no flags by default, or remove default value for simulation_flags altogether, as in simulateTransactions.

I'd lean towards sending without flags by default, but up to you really

Reply:
> hmm, that's weird. was this checked with some newer nodes' releases?

Checked this now. Can't test all of it, since we don't have goerli testnet funds, but invoke and deploy account seem to work fine on goerli integraiton and sepolia integration as of now. 

> Do you happen to know if that was by design, or by mistake? 

Query versions should work, so probably by mistake. The problem was also present on devnet and now seems to be resolved.

Considering this, it might be actually reasonable to not `skipValidate` by default. Although the fees are not that different, and that would allow sending unsigned transactions.